build.sbt

```diff
@@ -18,22 +11,26 @@ crossScalaVersions         := {
     sys.error(s"don't know what Scala versions to build on $java")
 }
 
-//reenable -Xfatal-warnings?
-scalacOptions             ++= "-deprecation:false -feature -Xlint:-stars-align,-nullary-unit,_".split("\\s+").to[Seq]
-
-scalacOptions in Test      += "-Xxml:coalescing"
-
-// important!! must come here (why?)
-scalaModuleOsgiSettings
-
-OsgiKeys.exportPackage := Seq(s"scala.xml.*;version=${version.value}")
-
-libraryDependencies += "junit" % "junit" % "4.11" % "test"
-
-libraryDependencies += "com.novocode" % "junit-interface" % "0.10" % "test"
-
-//// testing:
-// used in CompilerErrors test
-libraryDependencies += ("org.scala-lang" % "scala-compiler" % scalaVersion.value % "test").exclude("org.scala-lang.modules", s"scala-xml*")
-
-mimaPreviousVersion := Some("1.0.5")
+lazy val root = project.in(file("."))
+  .aggregate(xmlJS, xmlJVM)
+
+lazy val xml = crossProject
+  .crossType(CrossType.Full)
+  .settings(
+    name    := "scala-xml",
+    version := "1.0.6-SNAPSHOT",
+    scalacOptions         ++= "-deprecation:false -feature -Xlint:-stars-align,-nullary-unit,_".split("\\s+").to[Seq],
+    scalacOptions in Test  += "-Xxml:coalescing")
+  .jvmSettings(
+    scalaModuleSettings ++
+    scalaModuleOsgiSettings ++
+    List(
+      OsgiKeys.exportPackage := Seq(s"scala.xml.*;version=${version.value}"),
+      libraryDependencies += "junit" % "junit" % "4.11" % "test",
+      libraryDependencies += "com.novocode" % "junit-interface" % "0.10" % "test",
+      libraryDependencies += ("org.scala-lang" % "scala-compiler" % scalaVersion.value % "test").exclude("org.scala-lang.modules", s"scala-xml*"),
+      mimaPreviousVersion := Some("1.0.5")): _*)
+  .jsConfigure(_.enablePlugins(ScalaJSJUnitPlugin))
+
+lazy val xmlJVM = xml.jvm
+lazy val xmlJS = xml.js
```

Comment:
Will the `publish-signed` task work with the new changes to SBT?  The admin/build.sh file is run by Travis, and when it's for a Git tag, it will copy the admin/publish-settings.sbt file to the root directory.  Looks like it will cause SBT to fail to load:

```
scala-xml/publish-settings.sbt:3: error: not found: value pgpPassphrase
pgpPassphrase := Some(env("PGP_PASSPHRASE").toArray)
^
```